src/vivarium_csu_alzheimers/components/population.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+import pandas as pd
+from vivarium import Component
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import SimulantData
+from vivarium_public_health import utilities
+from vivarium_public_health.population import ScaledPopulation
+
+from vivarium_csu_alzheimers.constants import data_keys
+from vivarium_csu_alzheimers.constants.metadata import ARTIFACT_INDEX_COLUMNS
+
+
+class AlzheimersPopulation(ScaledPopulation):
+    def setup(self, builder: Builder) -> None:
+        super().setup(builder)
+        self.key_columns = builder.configuration.randomness.key_columns
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        if pop_data.user_data.get("sim_state") != "time_step":
+            super().on_initialize_simulants(pop_data)
+            return
+
+        # get pop_data.user_data["demographic_counts"] if it exists
+        if "demographic_counts" in pop_data.user_data:
+            # demographic_counts will be a pd.Series
+            demographic_counts = pop_data.user_data["demographic_counts"]
+        else:
+            # throw an error if it does not exist
+            raise ValueError(
+                "AlzPop requires demographic_counts in pop_data.user_data. "
+                "Please provide it in the user_data dictionary."
+            )
+
+        # Verify pop_data and demographic_counts are compatible
+        if not len(pop_data.index) == demographic_counts.sum():
+            raise ValueError(
+                "The index of pop_data does not match the index of demographic_counts. "
+                "Please ensure they are the same."
+            )
+
+        # Update metadata for each demographic group
+        new_simulants = pd.DataFrame(index=pop_data.index)
+        groups_to_add = demographic_counts[demographic_counts > 0]
+        start = pop_data.index.min()
+        for idx, row in groups_to_add.items():
+            sex, age_lower, age_upper = idx
+            value = row
+            group_idx = pd.Index(list(range(start, start + value)))
+            # This will always happen in the order of female then male and youngest to oldest
+            new_simulants.loc[group_idx, "sex"] = sex
+            age_draws = self.randomness["age_smoothing_age_bounds"].get_draw(
+                group_idx, additional_key=f"{sex}_{age_lower}_{age_upper}"
+            )
+            new_simulants.loc[group_idx, "age"] = age_lower + age_draws * (
+                age_upper - age_lower
+            )
+            start = group_idx.max() + 1
```

Comment:
Isn't this the same as `start + value`? I think that's a bit more intuitive personally, but this is also right.

Reply:
I think you are right.